code/cfile/cfile.cpp

```diff
@@ -593,6 +593,24 @@ int cf_rename(const char *old_name, const char *name, int dir_type)
 
 }
 
+static void mkdir_recursive(const char *dir) {
+    char tmp[256];
+    char *p = NULL;
+    size_t len;
+
+    snprintf(tmp, sizeof(tmp), "%s", dir);
+    len = strlen(tmp);
+    if (tmp[len - 1] == DIR_SEPARATOR_CHAR)
```

Comment:
I think it would be better to explicitly check both `/` and `\` to make sure that paths which use `/` don't break on Windows. This is not a problem in the current code because the directory table uses `DIR_SEPARATOR_CHAR` but it would avoid the error if the function is used differently in the future.